### PR TITLE
Added `baseURL` to the Azure OpenAI Provider

### PR DESCRIPTION
**Pull request by karl-richter**
Implementation of #2312 

### Main Changes
- Added `baseURL` ~~and `apiVersion`~~ as optional parameters to `AzureOpenAIProviderSettings`
- Added two test cases for the change

### Secondary Changes
~~- Added two parameters to the `loadSetting` function in `provider-utils`. The `defaultValue` param is needed to handle the fallback value for the `apiVersion` parameter. The `failOnMissing` param is needed to handle optional parameters in the Provider. It is used to check whether a `baseURL` is provided.~~
- Added a `getRequestUrl` function to the `JsonTestServer`. This is needed in the provider test to ensure that the `baseURL` is properly set.

### CODE REVIEW

**Comment by karl-richter** [comment]
Hi @lgrammel & others, I opened the issue #2312 earlier but now just went ahead and added the parameters to the Azure OpenAI provider in my fork. Please let me know of the change is intended / aligns with your plans & what the next steps would be to proceed with the review. Thanks!

**Comment by lgrammel** [comment]
@karl-richter I'm in favor of including the baseUrl. However, I have concerns re the api version. It should not be changed since our parsing relies on a specific api version. Instead we should update the version in the SDK. can you open a separate PR or remove the version part?

**Comment by karl-richter** [comment]
Thanks for the feedback @lgrammel! I have removed the `apiVersion` from the PR, I agree that the API version might have some downstream effects that should be handled respectively by the SDK. I also made the additional `failOnMissing` arg in the `loadSetting` function properly optional, so the build test should no longer fail. Could you trigger the Cloud tests again?

File: packages/azure/src/azure-openai-provider.ts [hunk 1]
```diff
@@ -101,8 +107,27 @@ export function createAzure(
...
     });
 
+  const checkIfBaseUrlIsProvided = () =>
+    loadSetting({
+      settingValue: options.baseURL,
+      settingName: 'baseURL',
+      environmentVariableName: 'AZURE_BASE_URL',
+      description: 'Azure OpenAI base URL',
+      failOnMissing: false,
+    });
```
Comment (lgrammel) [inline packages/azure/src/azure-openai-provider.ts:117]: I would much prefer if we can only pass in the base URL through options for now (no env param loading). This means all the "failOnMissing" etc is not required.

The env loading is reserved for only the super-common options to get started. The expectation is to use options once the configuration gets more complex.

Comment (lgrammel) [reply packages/azure/src/azure-openai-provider.ts:117]: (we need to document, test, and maintain this code and my preference is to keep it as simple as possible)

Comment (karl-richter) [reply packages/azure/src/azure-openai-provider.ts:117]: Alright, we can do that! I agree that this is simpler and less prone.

Comment (karl-richter) [reply packages/azure/src/azure-openai-provider.ts]: I could also remove the small test I added for the provider to check that the `baseURL` param works. Should we keep that or also remove it if we say that `baseURL` is not a core parameter?

Comment (lgrammel) [reply packages/azure/src/azure-openai-provider.ts]: the test is great! I'll continue on the PR tomorrow and add docs & changelog. Thank you!

**Comment by lgrammel** [comment]
continued in https://github.com/vercel/ai/pull/2337

**Pull request closed without merging**